src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -0,0 +1,445 @@
+                symbolValidator: (m) =>
+                {
+                    var result = (PEMethodSymbol)m.GlobalNamespace.GetTypeMember("I1").GetMember("M1");
+
+                    Assert.True(m1.IsMetadataVirtual());
+                    Assert.False(m1.IsAbstract);
+                    Assert.True(m1.IsVirtual);
+
+                    int rva;
+                    ((PEModuleSymbol)m).Module.GetMethodDefPropsOrThrow(result.Handle, out _, out _, out _, out rva);
+                    Assert.NotEqual(0, rva);
+                });
+
+            var source2 =
+@"
+class Test2 : I1
+{}
+";
+
+            var compilation2 = CreateCompilationWithMscorlib(source2, new[] { compilation1.ToMetadataReference() }, options: TestOptions.DebugDll);
+            Assert.True(compilation2.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            m1 = compilation2.GetMember<MethodSymbol>("I1.M1");
+            var test2 = compilation2.GetTypeByMetadataName("Test2");
+
+            Assert.Same(m1, test2.FindImplementationForInterfaceMember(m1));
+
+            compilation2.VerifyDiagnostics();
+            CompileAndVerify(compilation2, verify: false);
+
+            var compilation3 = CreateCompilationWithMscorlib(source2, new[] { compilation1.EmitToImageReference() }, options: TestOptions.DebugDll);
+            Assert.True(compilation3.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            m1 = compilation3.GetMember<MethodSymbol>("I1.M1");
+            test2 = compilation3.GetTypeByMetadataName("Test2");
+
+            Assert.Same(m1, test2.FindImplementationForInterfaceMember(m1));
+
+            compilation3.VerifyDiagnostics();
+            CompileAndVerify(compilation3, verify: false);
+        }
+
+        [Fact]
+        public void MethodImplementation_02()
+        {
+            var source1 =
+@"
+interface I1
+{
+    void M1() {}
+    void M2() {}
+}
+
+class Base
+{
+    void M1() { }
+}
+
+class Derived : Base, I1
+{
+    void M2() { }
+}
+
+class Test : I1 {}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics(
+                // (13,23): error CS0737: 'Derived' does not implement interface member 'I1.M2()'. 'Derived.M2()' cannot implement an interface member because it is not public.
+                // class Derived : Base, I1
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberNotPublic, "I1").WithArguments("Derived", "I1.M2()", "Derived.M2()").WithLocation(13, 23),
+                // (13,23): error CS0737: 'Derived' does not implement interface member 'I1.M1()'. 'Base.M1()' cannot implement an interface member because it is not public.
```

Comment:
This diagnostic does not appear to be correct. Derived does implement I1.M2() by inheriting the default implementation.